third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -191,14 +191,16 @@ template <typename FPType> struct FPTypeInfo {
 template <typename ConvertOp>
 static SmallVector<Value>
 cvtScalePkUpcastFromFp8(Location loc, ConversionPatternRewriter &rewriter,
-                        Value v0, Value v1) {
+                        const SmallVector<Value> v) {
+  assert(v.size() == 4);
   auto b = TritonLLVMOpBuilder(loc, rewriter);
   auto fp8x4VecTy = vec_ty(i8_ty, 4);
   Value fp8x4Vec = b.undef(fp8x4VecTy);
-  auto idx0 = b.i32_val(0);
-  auto idx1 = b.i32_val(1);
-  fp8x4Vec = b.insert_element(fp8x4VecTy, fp8x4Vec, v0, idx0);
-  fp8x4Vec = b.insert_element(fp8x4VecTy, fp8x4Vec, v1, idx1);
+  SmallVector<Value> idx(4);
```

Comment:
For this kind of fixed-size cases, good to just spell it out like `SmallVector<Value, 4>` to have all values allocated inline.